### PR TITLE
refac: Use JWT permission middleware.

### PR DESCRIPTION
**Pull request by defectiveAi**
Enabling permissions requires new middleware that can validate the updated token. This PR has to be completed simultaneously with other domains.

### CODE REVIEW

File: source/dotnet/Services/Infrastructure/Infrastructure.csproj [hunk 1]
```diff
@@ -29,7 +29,7 @@ limitations under the License.
     <ItemGroup>
```
Comment (defectiveAi) [inline source/dotnet/Services/Infrastructure/Infrastructure.csproj:29]: Review, but do not complete.

Comment (BjarkeMeier) [reply source/dotnet/Services/Infrastructure/Infrastructure.csproj:29]: I'd recommend changing the state to draft if the PR must not be merged yet.

File: source/dotnet/Services/DomainTests/Fixtures/UserTokenPermissionAugmentation.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        {
+            var actorId = await GetActorAsync(externalAccessToken);
+            return await AugmentTokenAsync(externalAccessToken, actorId);
+        }
+
+        private async Task<Guid> GetActorAsync(string externalToken)
+        {
+            using var response = await _tokenHttpClient.GetAsync($"user/actors?externalToken={externalToken}");
+            var actors = await response.Content.ReadFromJsonAsync<GetAssociatedUserActorsResponseDto>();
+            return actors!.ActorIds.First();
```
Comment (lasrinnil) [inline source/dotnet/Services/DomainTests/Fixtures/UserTokenPermissionAugmentation.cs:50]: Could we handle if the response content could not be parsed with a more explicit exception? Would be easier for potential debuggin

File: source/dotnet/Services/IntegrationTests/TestCommon/WebApi/BearerTokenTests.cs [hunk 3]
```diff
@@ -69,27 +62,11 @@ public async Task Request_WhenInvalidBearerToken_Returns401Unauthorized()
...
 
         // Act
         var response = await client.GetAsync(BaseUrl);
 
         // Assert
         response.StatusCode.Should().Be(HttpStatusCode.Unauthorized);
     }
-
-    [Fact]
-    public async Task Request_WhenValidBearerToken_DoesNotReturn401Unauthorized()
```
Comment (lasrinnil) [inline source/dotnet/Services/IntegrationTests/TestCommon/WebApi/BearerTokenTests.cs:82]: Is the token always valid?

File: source/dotnet/Services/WebApi/Startup.cs [hunk 4]
```diff
@@ -39,10 +38,6 @@ public void ConfigureServices(IServiceCollection services)
         services.AddControllers();
         services.AddEndpointsApiExplorer();
         services.AddSwaggerGen();
-        if (!Environment.IsDevelopment())
```
Comment (lasrinnil) [inline source/dotnet/Services/WebApi/Startup.cs:42]: nice

**Review by lasrinnil — APPROVED** [review]

File: source/dotnet/Services/IntegrationTests/TestCommon/WebApi/BearerTokenTests.cs (hunk 3 above)
Comment (defectiveAi) [reply source/dotnet/Services/IntegrationTests/TestCommon/WebApi/BearerTokenTests.cs:82]: No. The old test "validated" an invalid token 'Bearer xxx' by faking the implementation. This is no longer possible, as the validation code is no longer ours, but is supplied by .NET; and .NET will correctly reject such tokens.

File: source/dotnet/Services/DomainTests/Fixtures/UserTokenPermissionAugmentation.cs (hunk 2 above)
Comment (defectiveAi) [reply source/dotnet/Services/DomainTests/Fixtures/UserTokenPermissionAugmentation.cs:50]: Fixed

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**